miden/tests/integration/operations/stack_ops.rs

```diff
@@ -14,17 +197,462 @@ fn swapdw() {
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swap_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swap";
+        let mut expected_values = test_values.to_vec();
+        expected_values.reverse();
+        expected_values.swap(0, 1);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 1_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("swap.{n}");
+        let mut expected_values = test_values.to_vec();
+        expected_values.reverse();
+        expected_values.swap(0, n);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swapw";
+        let mut expected_values = test_values.to_vec();
+        let mut a = expected_values.split_off(WORD_LEN*3);
+        let mut b = expected_values.split_off(WORD_LEN*2);
+        expected_values.append(&mut a);
+        expected_values.append(&mut b);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapwn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 1_usize..WORD_LEN) {
+        let asm_op = format!("swapw.{n}");
+        let mut expected_values = test_values.to_vec();
+        let mut a = expected_values.split_off(WORD_LEN*(MIN_STACK_DEPTH/WORD_LEN-n-1));
+        let mut b = a.split_off(WORD_LEN);
+        let mut c = b.split_off(b.len()-WORD_LEN);
+        expected_values.append(&mut c);
+        expected_values.append(&mut b);
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn swapdw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "swapdw";
+        let mut expected_values = test_values[..WORD_LEN*2].to_vec();
+        let mut b = test_values[WORD_LEN*2..].to_vec();
+        expected_values.reverse();
+        b.reverse();
+        expected_values.append(&mut b);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn movup_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), movup_idx in 2_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("movup.{movup_idx}");
+        let mut expected_values = test_values.to_vec();
+        let movup_value = expected_values[MIN_STACK_DEPTH - movup_idx - 1];
+        expected_values.remove(MIN_STACK_DEPTH - movup_idx - 1);
```

Comment:
For readability, I'd prefer if `MIN_STACK_DEPTH - movup_idx - 1` were defined as a variable above, especially since it's used twice.